src/common/meta/src/wal_options_allocator/kafka/topic_manager.rs

```diff
@@ -108,6 +109,19 @@ impl TopicManager {
         Ok(())
     }
 
+    async fn resolve_broker_endpoint(broker_endpoint: &str) -> Result<String> {
+        let ip = net::lookup_host(broker_endpoint)
+            .await
+            .with_context(|_| ResolveKafkaEndpointSnafu {
+                broker_endpoint: broker_endpoint.to_string(),
+            })?
+            // Not sure if we should filter out ipv6 addresses
+            .find(|addr| addr.is_ipv4())
+            .with_context(|| EndpointIpNotFoundSnafu {
+                broker_endpoint: broker_endpoint.to_string(),
+            })?;
+        Ok(ip.to_string())
+    }
```

Comment:
```suggestion
    async fn resolve_broker_endpoint(broker_endpoint: &str) -> Result<String> {
        let ip = net::lookup_host(broker_endpoint)
            .await
            .with_context(|_| ResolveKafkaEndpointSnafu {
                broker_endpoint: broker_endpoint.to_string(),
            })?
            // Not sure if we should filter out ipv6 addresses
            .find(|addr| addr.is_ipv4())
            .with_context(|| EndpointIpNotFoundSnafu {
                broker_endpoint: broker_endpoint.to_string(),
            })?;
        Ok(ip.to_string())
    }
```

It's better to move this piece of code into the `greptimedb/src/common/wal/lib.rs` file to avoid duplication. A new file `error.rs` is necessary to be composed for error handling. You can follow other `error.rs` files.

Reply:
A unit test is necessary to prove this function works as expected.

Reply:
I've added mock test. And I think we can assume that `tokio::net::lookup_host` work as expected.